### PR TITLE
Split WalletService into BitcoinWalletService and LiquidWalletService

### PR DESCRIPTION
**Pull request by alvasw**
- Add RpcConfig verification
- Split RpcConfig into DaemonRpcConfig and WalletRpcConfig
- Add Wallet submenu in UI for both cryptocurrencies
- Split WalletStore into BitcoinWalletStore and LiquidWalletStore
- Persist RPC config after verifying whether they are correct

### CODE REVIEW

File: wallets/src/main/java/bisq/wallets/RpcConfig.java [hunk 1]
```diff
@@ -15,62 +15,68 @@
...
-    public WalletStore() {
-    }
-
-    public WalletStore(BitcoinWalletStore bitcoinWalletStore, LiquidWalletStore liquidWalletStore) {
-        this.bitcoinWalletStore = bitcoinWalletStore;
-        this.liquidWalletStore = liquidWalletStore;
-    }
+@Builder
+@Getter
+public class RpcConfig implements PersistableStore<RpcConfig> {
```
Comment (chimp1984) [inline wallets/src/main/java/bisq/wallets/RpcConfig.java]: I would prefer to stick with the convention to call classes used for persistence store -> RpcConfigStore.

File: wallets/src/main/java/bisq/wallets/stores/BitcoinWalletStore.java [hunk 2]
```diff
@@ -21,34 +21,46 @@
...
 
 import java.util.HashSet;
+import java.util.Optional;
 import java.util.Set;
 
 @Slf4j
 public class BitcoinWalletStore implements PersistableStore<BitcoinWalletStore> {
+    @Getter
+    @Setter
+    private Optional<RpcConfig> rpcConfig = Optional.empty();
```
Comment (chimp1984) [inline wallets/src/main/java/bisq/wallets/stores/BitcoinWalletStore.java:38]: Ah the RpcConfig is using inside the walletstores. In that case ignore my previous comment, but I think RpcConfig does not require to implement PersistableStore if it is not stored itself. PersistableStore is used for the classes which used for the storage file.

**Review by chimp1984 — CHANGES_REQUESTED** [review]

**Comment by chimp1984** [comment]
Does the initialize call on both wallets cause some delay/resources (beside creating directories)?
As those wallets are optional I guess we should delay initialisation to the moment when the user decides to select a wallet for usage. We can leave it for the moment and once the UI is more final see how to deal with it best.

File: wallets/src/main/java/bisq/wallets/RpcConfig.java [hunk 3]
```diff
@@ -20,15 +20,29 @@
...
-
 @Builder
 @Getter
-public class WalletConfig {
-    private final WalletBackend walletBackend;
-    private final String hostname;
-    private final int port;
-    private final String user;
-    private final String password;
+public class RpcConfig {
```
Comment (chimp1984) [inline wallets/src/main/java/bisq/wallets/RpcConfig.java]: Should `implements Proto` to mark it as class used for protobuf

File: wallets/src/main/java/bisq/wallets/RpcConfig.java (hunk 1 above)
Comment (alvasw) [reply wallets/src/main/java/bisq/wallets/RpcConfig.java]: You're right the `RpcConfig` does not need to implement `PersistableStore`. I updated the PR.

File: wallets/src/main/java/bisq/wallets/stores/BitcoinWalletStore.java (hunk 2 above)
Comment (alvasw) [reply wallets/src/main/java/bisq/wallets/stores/BitcoinWalletStore.java:38]: You're right the `RpcConfig` does not need to implement `PersistableStore`. I updated the PR.

**Comment by alvasw** [comment]
> Does the initialize call on both wallets cause some delay/resources (beside creating directories)? As those wallets are optional I guess we should delay initialisation to the moment when the user decides to select a wallet for usage. We can leave it for the moment and once the UI is more final see how to deal with it best.

No, the initialize call does nothing except creating the directories. It only instantiates a wallet implementation if a persisted RpcConfig exists for a wallet. Otherwise, the UI later calls `initializeWallet` to instantiate a wallet implementation.

**Comment by alvasw** [comment]
I rebased this PR with main and resolved the merge conflict.

**Comment by chimp1984** [comment]
> I rebased this PR with main and resolved the merge conflict.

I think there is still an unresolved remark: `Should implements Proto to mark it as class used for protobuf` for RpcConfig

**Comment by alvasw** [comment]
> I think there is still an unresolved remark: `Should implements Proto to mark it as class used for protobuf` for RpcConfig

`RpcConfig` implements `Proto` now.

**Review by chimp1984 — APPROVED** [review]
utACK

File: desktop/src/main/java/bisq/desktop/primary/main/left/LeftNavView.java [hunk 4]
```diff
@@ -298,20 +311,30 @@ private void maybeAnimateMark() {
             double targetY = menuTop + selectedLeftNavButton.getBoundsInParent().getMinY();
             if (selectedLeftNavButton instanceof LeftNavSubButton) {
                 targetY += tradeSubMenuItems.getLayoutY();
+                targetY += walletSubMenuItems.getLayoutY();
```
Comment (ghost) [inline desktop/src/main/java/bisq/desktop/primary/main/left/LeftNavView.java:314]: @alvasw I see a few UI issues with wallet submenu:
- when Trade menu is selected Wallet menu is incorrectly highlighted
- when any protocol from Trade submenu is selected then green marker moves too low

Preview:
navigation-issue

Comment (alvasw) [reply desktop/src/main/java/bisq/desktop/primary/main/left/LeftNavView.java:314]: Oops, I just saw that. You're right. Can you fix it? I have a lot of other things to do at the moment.

Comment (ghost) [reply desktop/src/main/java/bisq/desktop/primary/main/left/LeftNavView.java:314]: Yes - I'll be updating this menu soon.

Comment (alvasw) [reply desktop/src/main/java/bisq/desktop/primary/main/left/LeftNavView.java:314]: Thank you!

**Pull request merged**